components/datetime/src/neo.rs

```diff
@@ -147,6 +147,43 @@ macro_rules! gen_any_buffer_constructors_with_external_loader {
     };
 }
 
+/// Options bag for datetime formatting.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+#[non_exhaustive]
+pub struct NeoOptions<R: DateTimeMarkers> {
+    /// The desired length of the formatted string,
+    /// if required for the chosen field set.
+    ///
+    /// See [`NeoSkeletonLength`].
+    pub length: R::LengthOption,
+}
+
+impl<R> From<NeoSkeletonLength> for NeoOptions<R>
+where
+    R: DateTimeMarkers,
+    R::LengthOption: From<NeoSkeletonLength>,
+{
+    #[inline]
+    fn from(value: NeoSkeletonLength) -> Self {
+        NeoOptions {
+            length: value.into(),
+        }
+    }
+}
+
+impl<R> From<()> for NeoOptions<R>
```

Comment:
This is basically `Default`, but bad

Reply:
I see `impl From<()>` as meaning "this is a singleton type and this gets the only value for it" and `impl Default` as meaning "this type has many possible values and this gets the the emptiest such value".

Reply:
I don't

Reply:
I can use `Default::default()` now that I have a concrete type instead of `impl Into`. Done.